src/main/java/net/dv8tion/jda/core/entities/MessageChannel.java

```diff
@@ -466,8 +473,15 @@ protected void handleResponse(Response response, Request request)
 
         if (message != null)
         {
-            body.field("content", message.getRawContent());
-            body.field("tts", message.isTTS());
+            if (message.getEmbeds().isEmpty())
+            {
+                body.field("content", message.getRawContent());
+                body.field("tts", message.isTTS());
+            }
+            else
+            {
+                body.field("payload_json", ((MessageImpl) message).toJSONObject().toString());
```

Comment:
we could switch completely, but this way is a bit more efficient (no json object creation and less total data to send)